src/app/projects/components/project-settings/components/embeddings/embeddings.component.ts

```diff
@@ -65,18 +86,43 @@ export class EmbeddingsComponent implements OnInit, OnDestroy, OnChanges {
       }));
   }
 
+  prepareEmbeddingPlatforms() {
+    if (this.organization.gdprCompliant && this.isManaged) {
+      this.embeddingPlatforms = this.embeddingPlatforms.filter((platform) => platform.gdprCompliant === true);
+    }
+    this.embeddingPlatforms.forEach((platform: EmbeddingPlatform) => {
+      platform.name = platformNamesDict[platform.platform];
+    });
+  }
+
   checkForceHiddenHandles() {
-    const granularity = this.settingModals.embedding.create.embeddingCreationFormGroup.get('granularity').value;
-    const attId = this.settingModals.embedding.create.embeddingCreationFormGroup.get('targetAttribute').value;
-    const suggestionList = this.embeddingHandles[attId];
+    const form = this.settingModals.embedding.create.embeddingCreationFormGroup;
+    const granularity = form.get('granularity').value;
+    const attId = form.get('targetAttribute').value;
+    const platform = form.get('platform').value;
+    let suggestionList = this.embeddingHandlesCopy[attId];
+    if (platform == PlatformType.PYTHON) {
+      suggestionList = suggestionList.filter(e => e.configString == 'bag-of-words' || e.configString == 'bag-of-characters' || e.configString == 'tf-idf');
+      this.embeddingHandles[attId] = suggestionList;
+    } else if (platform == PlatformType.OPEN_AI) {
+      suggestionList = suggestionList.filter(e => e.configString == 'text-embedding-ada-002');
+      this.embeddingHandles[attId] = suggestionList;
+    } else if (platform == PlatformType.HUGGING_FACE) {
+      this.embeddingHandles[attId] = this.embeddingHandlesCopy[attId];
+      this.checkModelDownloaded();
+    }
     for (let element of suggestionList) {
       element.forceHidden = true;
       const parseEl = JSON.parse(element.applicability);
-      if ((granularity == 'ON_ATTRIBUTE' && parseEl.attribute)
-        || (granularity == 'ON_TOKEN' && parseEl.token)) {
+      if ((granularity == EmbeddingType.ON_ATTRIBUTE && parseEl.attribute)
+        || (granularity == EmbeddingType.ON_TOKEN && parseEl.token)) {
         element.forceHidden = false;
       }
     }
+    this.selectedPlatform = this.embeddingPlatforms.find((p: EmbeddingPlatform) => p.platform == platform);
+    form.get('model').setValue(null);
+    form.get('apiToken').setValue(null);
+    form.get('termsAccepted').setValue(false);
```

Comment:
not sure if i understand everything correctly. but when I already have a platform selected switching between model shouldn't result in clearing evything.

So not every change results the fields. Changing the platform, however, does.

Reply:
Yes switching between models does not result in clearing everything. Currently it clears the selected options when switching between platform and granularity. Do we want this for granularity?

Reply:
misunderstanding. the method is call checkForceHiddenHandles. i thought this was called on keydown of the combination boxes (dropdown with search). Thats the hidden flag, not the forcehidden one. Sorry :D can be resolved